packages/framework/src/Component/Elasticsearch/ElasticsearchStructureManager.php

```diff
@@ -102,4 +242,104 @@ protected function getDefinition(int $domainId, string $index): array
+        ];
+        $currentIndexDefinition['settings']['analysis']['filter']['edge_ngram']['max_gram'] =
+            (int)$currentIndexDefinition['settings']['analysis']['filter']['edge_ngram']['max_gram'];
+        $currentIndexDefinition['settings']['analysis']['filter']['edge_ngram']['min_gram'] =
+            (int)$currentIndexDefinition['settings']['analysis']['filter']['edge_ngram']['min_gram'];
+        unset(
+            $currentIndexDefinition['aliases'],
+            $currentIndexDefinition['settings']['provided_name'],
+            $currentIndexDefinition['settings']['creation_date'],
+            $currentIndexDefinition['settings']['uuid'],
+            $currentIndexDefinition['settings']['version'],
+            $currentIndexDefinition['settings']['number_of_replicas'],
+            $currentIndexDefinition['settings']['number_of_shards']
+        );
+        return $currentIndexDefinition;
+    }
+
+    /**
+     * @param int $domainId
+     * @param string $index
+     */
+    public function createIndexIfNecessary(int $domainId, string $index)
+    {
+        if ($this->isNecessaryToUpdateStructure($domainId, $index)) {
+            $this->createIndex($domainId, $index);
+        }
+    }
+
+    /**
+     * @param int $domainId
+     * @param string $index
+     */
+    public function createIndexAndReindexIfNecessary(int $domainId, string $index)
+    {
+        if ($this->isNecessaryToUpdateStructure($domainId, $index)) {
+            $this->createIndex($domainId, $index, false);
+            $this->reindex($domainId, $index);
+            $this->deleteNotUsedIndexesOfCurrentAlias($domainId, $index);
+            $this->createAliasForCurrentIndex($domainId, $index);
+        }
+    }
+
+    /**
+     * @param int $domainId
+     * @param string $index
+     * @return bool
+     */
+    protected function isNecessaryToUpdateStructure(int $domainId, string $index)
+    {
+        $definition = $this->getDefinition($domainId, $index);
+        $indexes = $this->client->indices();
+
+        $aliasName = $this->getAliasName($domainId, $index);
+        if ($indexes->existsAlias(['name' => $aliasName])) {
+            $aliases = $indexes->getAlias(['name' => $aliasName]);
+            foreach (array_keys($aliases) as $indexName) {
+                if ($indexes->exists(['index' => $indexName])) {
+                    $currentIndex = $indexes->get(['index' => $indexName]);
+                    $currentIndexDefinition = $this->prepareCurrentIndexDefinition($currentIndex);
+
+                    $this->recursiveArraySort($currentIndexDefinition);
+                    $this->recursiveArraySort($definition);
+
+                    if ($currentIndexDefinition !== $definition) {
+                        return true;
+                    }
+                }
+            }
+        } else {
+            return true;
+        }
+
+        return false;
```

Comment:
this line will never executed as there is `else` 4 lines above. I recommend to remove the `else` statement completely and keep here only `return true`